packages/venia-ui/lib/components/Icon/icon.js

```diff
@@ -1,39 +1,41 @@
-import React, { Component } from 'react';
-import { func, number, object, oneOfType, shape, string } from 'prop-types';
+import React from 'react';
+import { func, number, shape, string } from 'prop-types';
 
-import classify from '../../classify';
+import { mergeClasses } from '../../classify';
 import defaultClasses from './icon.css';
 
-/**
- * The Icon component allows us to wrap each icon with some default styling.
- */
-class Icon extends Component {
-    static propTypes = {
-        classes: shape({
-            root: string
-        }),
-        size: number,
-        attrs: object,
-        src: oneOfType([func, shape({ render: func.isRequired })]).isRequired
-    };
+const Icon = props => {
+    // destructure `propClasses` to exclude it from `restProps`
+    const {
+        attrs,
+        classes: propClasses,
+        size,
+        src: Component,
+        ...restProps
+    } = props;
+    const { width, ...restAttrs } = attrs || {};
+    const classes = mergeClasses(defaultClasses, propClasses);
 
-    render() {
-        const {
-            attrs: { width, ...restAttrs } = {},
-            size,
-            classes,
-            src: IconComponent
-        } = this.props;
+    return (
+        <span className={classes.root} {...restProps}>
+            <Component
+                className={classes.icon}
+                size={size || width}
+                {...restAttrs}
+            />
+        </span>
+    );
+};
 
-        // Permit both prop styles:
-        // <Icon src={Foo} attrs={{ width: 18 }} />
-        // <Icon src={Foo} size={18} />
-        return (
-            <span className={classes.root}>
-                <IconComponent size={size || width} {...restAttrs} />
-            </span>
-        );
-    }
-}
+export default Icon;
 
-export default classify(defaultClasses)(Icon);
+Icon.propTypes = {
+    attrs: shape({}),
+    classes: shape({
+        'icon--empty': string,
+        'icon--filled': string,
```

Comment:
I don't think this is accurate - we aren't passing `icon--empty` or `icon--filled` with `classes`. We apply one or the other to the `Icon` itself via `className`. It just happens to get spread onto the Icon through `restProps`.

Reply:
Oh, I see what I did. I put the new classnames in the prop types for `Icon` rather than `CartTrigger`. I'll fix that.

As for `className` as a prop, you're right. The `classify` HOC allowed us to pass `className` as a shortcut override for `classes.root`, but its `mergeClasses` utility is simpler and doesn't have that feature. I'll update `CartTrigger` to override `classes.root` more explicitly.